File: Easy/JAVASCRIPT/ProductSum.js
const productSum = (array) => {
    return productSumHelper(1, array);
}

const productSumHelper = (factor, array) => {
    sum = 0;
    for (const element of array) {
        // Check if the element is an integer or a list
        if (Array.isArray(element))
            sum += productSumHelper(factor + 1, element);
        else
            sum += element;
    }
    return sum * factor;
}

const specialArray = [5, 2, [7, -1], 3, [6, [-13, 8], 4]];
const result = productSum(specialArray);
console.log(result);

File: Easy/C#/ProductSum.cs
using System.Collections.Generic;

class ProductSum
{
    public static int ProductSummation(List<object> array)
    {
        return ProductSummationHelper(1, array);
    }

    // O(n) n is the number of the total number of elements.
    // O(D) depth of special array in arrays.
    public static int ProductSummationHelper(int factor, List<object> array)
    {
        int sum = 0;
        foreach (object element in array)
        {
            if (element is IList<object>)
                sum += ProductSummationHelper(factor + 1, (List<object>)element);
            else
                sum += (int)element;
        }
        return sum * factor;
    }

}